### PR TITLE
Add mirror csi images code

### PR DESCRIPTION
**Pull request by yangchiu**
Add mirror csi images code

For [#3056](https://github.com/longhorn/longhorn/issues/3056)

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: mirror_csi_images/scripts/image-mirror.sh [hunk 1]
```diff
@@ -0,0 +1,211 @@
+#!/bin/bash
+
+set -uo pipefail
+export DOCKER_CLI_EXPERIMENTAL="enabled"
+
+ARCH_LIST="amd64 arm64 arm s390x"
```
Comment (innobead) [inline mirror_csi_images/scripts/image-mirror.sh:6]: Does CSI upstream support `arm` as well?

File: mirror_csi_images/scripts/publish.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+IFS=, read -ra CSI_IMAGES_ARR <<< "${CSI_IMAGES}"
+for CSI_IMAGE in "${CSI_IMAGES_ARR[@]}"; do
+  IFS=: read -ra IMAGE_TAG_PAIR <<< "$CSI_IMAGE"
+  echo "k8s.gcr.io/sig-storage/${IMAGE_TAG_PAIR[0]}" "longhornio/${IMAGE_TAG_PAIR[0]}" "${IMAGE_TAG_PAIR[1]}" >> "${PWD}/infile"
+done
+
+docker login -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}"
+
+source "${PWD}/mirror_csi_images/scripts/image-mirror.sh" "${PWD}/infile"
```
Comment (innobead) [inline mirror_csi_images/scripts/publish.sh:13]: Do we need source?

Can we make `"${PWD}/infile"` as a variable at the top?

**Review by innobead — CHANGES_REQUESTED** [review]

File: mirror_csi_images/scripts/image-mirror.sh (hunk 1 above)
Comment (yangchiu) [reply mirror_csi_images/scripts/image-mirror.sh:6]: it varies from different packages and different versions (tags).
the complete CSI registry can be found here: https://console.cloud.google.com/gcr/images/k8s-staging-sig-storage

Comment (innobead) [reply mirror_csi_images/scripts/image-mirror.sh:6]: Sounds good.

**Review by innobead — APPROVED** [review]

**Pull request merged**